executor/modes/local/runner.py

```
@@ -205,6 +212,78 @@ async def cancel_task(self, task_id: int) -> None:
+            # Cancel the task if agent supports it
+            if self.current_agent and hasattr(self.current_agent, "cancel_run"):
+                self.current_agent.cancel_run()
+
+            # Cleanup agent resources
+            if self.current_agent and hasattr(self.current_agent, "cleanup"):
+                try:
+                    self.current_agent.cleanup()
+                    logger.info(
+                        f"Cleaned up current agent resources for task {task_id}"
+                    )
+                except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)